src/main/java/com/hierynomus/smbj/utils/SmbFiles.java

```diff
@@ -64,4 +74,31 @@ public static int copy(File source, DiskShare share, String destPath, boolean ov
         return r;
     }
 
+    /**
+     * Create a set of nested sub-directories in the given path, for example, 2345 \ 3456 \ 4453 \ 123123.txt
+     */
+    public void mkdirs(DiskShare diskShare, String path) throws SMBApiException {
+        String currPath = path;
+        StringBuilder currDir = new StringBuilder();
+        while (path.startsWith("\\"))
+            currPath = currPath.substring(1, currPath.length() - 1 );
+
+        if (currPath.indexOf( "\\") <= 0 && !diskShare.folderExists(currPath)) {
+            diskShare.mkdir(currPath);
+        }   else {
+            for (String dir: currPath.split(String.valueOf("\\\\"))) {
+                currDir = currDir.append(dir).append("\\");
+                if (currDir != null && currDir.length() > 0 && !diskShare.folderExists(currDir.toString())) {
+                    Directory fileHandle = diskShare.openDirectory(currDir.toString(),
```

Comment:
Why not directly call `diskShare.mkdir` here instead of duplicating that code?

Reply:
fixed